File: leetcode/383.py
"""
    File: 383.py
    Title: Ransom Note
    Difficulty: Easy
"""

from collections import Counter


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        s = Counter(magazine)
        for c in ransomNote:
            if s[c] == 0:
                return False
            s[c] -= 1
        return True

File: leetcode/383.go
// File: 383.go
// Title: Ransom Note
// Difficulty: Easy

package leetcode

func canConstruct(ransomNote string, magazine string) bool {
	m := make(map[rune]int)
	for _, r := range magazine {
		m[r] += 1
	}
	for _, r := range ransomNote {
		m[r] -= 1
		if m[r] < 0 {
			return false
		}
	}
	return true
}
